File: src/taburet/transactions/_design/transactions/views/transactions/map.js
function(doc) {
	if ( doc.doc_type == 'Transaction' ) {
		emit([2, doc.from_acc[doc.from_acc.length-1], doc.date[0], doc.date[1], doc.date[2]], null)
		emit([1, doc.to_acc[doc.to_acc.length-1], doc.date[0], doc.date[1], doc.date[2]], null)
		emit([3, doc.from_acc[doc.from_acc.length-1], doc.date[0], doc.date[1], doc.date[2]], null)
		emit([3, doc.to_acc[doc.to_acc.length-1], doc.date[0], doc.date[1], doc.date[2]], null)
	}
}

File: src/taburet/transactions/_design/transactions/views/transactions/map.erl
fun({Doc}) ->
	case proplists:get_value(<<"doc_type">>, Doc) of
		<<"Transaction">> ->
			[Year, Month, Day | _] = proplists:get_value(<<"date">>, Doc),
			FromAccId = lists:last(proplists:get_value(<<"from_acc">>, Doc)),
			ToAccId = lists:last(proplists:get_value(<<"to_acc">>, Doc)),
			Emit([2, FromAccId, Year, Month, Day], null),
			Emit([1, ToAccId, Year, Month, Day], null),
			Emit([3, FromAccId, Year, Month, Day], null),
			Emit([3, ToAccId, Year, Month, Day], null);
		_ -> ok
	end
end.
